packages/runtime/src/helpers/functions.ts

```diff
@@ -165,18 +165,22 @@ export const setupImageFunction = async ({
 }
 
 /**
- * Look for API routes, and extract the config from the source file.
+ * Look for extended API routes (background and scheduled functions) and extract the config from the source file.
  */
-export const getApiRouteConfigs = async (publish: string, baseDir: string): Promise<Array<ApiRouteConfig>> => {
+export const getExtendedApiRouteConfigs = async (publish: string, baseDir: string): Promise<Array<ApiRouteConfig>> => {
   const pages = await readJSON(join(publish, 'server', 'pages-manifest.json'))
   const apiRoutes = Object.keys(pages).filter((page) => page.startsWith('/api/'))
   const pagesDir = join(baseDir, 'pages')
-  return Promise.all(
+
+  const settledApiRoutes = await Promise.all(
     apiRoutes.map(async (apiRoute) => {
       const filePath = getSourceFileForPage(apiRoute, pagesDir)
       return { route: apiRoute, config: await extractConfigFromFile(filePath), compiled: pages[apiRoute] }
     }),
   )
+
+  // We only want to return the API routes that are background or scheduled functions
+  return settledApiRoutes.filter((apiRoute) => apiRoute.config.type !== undefined)
```

Comment:
[pebble] I think it would be better to filter outside of this function, and return all of the configs here. There are scenarios where it would be useful to get the configs for other functions too, because it would let us find API routes that use the edge runtime.

Reply:
I had done it here because there appeared to be two places I needed to filter, but if we have other uses for this, I can filter at the call sites.